### PR TITLE
PUBDEV-7312 - AWS Auth with session tokens

### PR DESCRIPTION
**Pull request by Pscheidl**
https://0xdata.atlassian.net/browse/PUBDEV-7312

Marked for `3.28.0.4`

### CODE REVIEW

File: h2o-persist-s3/src/main/java/water/persist/PersistS3CredentialsV3.java [hunk 1]
```diff
@@ -11,4 +11,7 @@
     
     @API(required = true, direction = API.Direction.INPUT, level = API.Level.secondary, help = "S3 Secret Key")
     public  String secret_access_key;
+    
+    @API(required = true, direction = API.Direction.INPUT, level = API.Level.secondary, help = "S3 Session token")
```
Comment (michalkurka) [inline h2o-persist-s3/src/main/java/water/persist/PersistS3CredentialsV3.java]: required = true?

Comment (Pscheidl) [reply h2o-persist-s3/src/main/java/water/persist/PersistS3CredentialsV3.java]: I am just pushing it !

Comment (Pscheidl) [reply h2o-persist-s3/src/main/java/water/persist/PersistS3CredentialsV3.java]: Pushed. Thanks

File: h2o-persist-s3/src/main/java/water/persist/PersistS3.java [hunk 2]
```diff
@@ -95,8 +95,11 @@ public H2OAWSCredentialsProviderChain() {
     public AWSCredentials getCredentials() {
       final IcedS3Credentials s3Credentials = DKV.getGet(IcedS3Credentials.S3_CREDENTIALS_DKV_KEY);
 
-      if (s3Credentials != null && s3Credentials._secretKeyId != null && s3Credentials._secretAccessKey != null) {
```
Comment (honzasterba) [inline h2o-persist-s3/src/main/java/water/persist/PersistS3.java]: removed check `s3Credentials != null` on purspose?

Comment (Pscheidl) [reply h2o-persist-s3/src/main/java/water/persist/PersistS3.java]: No, this definitely has to be there !

Comment (Pscheidl) [reply h2o-persist-s3/src/main/java/water/persist/PersistS3.java]: Added. Thank you for catching this.

**Comment by Pscheidl** [comment]
Ok, tested manually, works properly.

Now let's wait for the tests.

**Pull request merged**